utils/heap/map_test.go

```diff
@@ -0,0 +1,96 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package heap
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestMap(t *testing.T) {
+	tests := []struct {
+		name     string
+		setup    func(h Map[int, int])
```

Comment:
nit: consider using Map[string, int] as a type. Just to avoid suggesting the key should be related to the entry position in the heap?